graphdatascience/error/deprecation_warning.py

```diff
@@ -0,0 +1,21 @@
+from functools import wraps
+from logging import warning
+from typing import Any, Callable, TypeVar, cast
+
+from ..caller_base import CallerBase
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+def deprecation_warning(new_procedure: str) -> Callable[[F], F]:
```

Comment:
Nice! :clap:

Reply:
I guess this function could also take arguments specifying for which server versions the warning should be displayed